VIPs/vip-7.md

```diff
@@ -0,0 +1,260 @@
+2. Minimize new user friction when creating a DID
+3. Maximising flexibility for advanced users
+4. Maximising network performance for end users
+
+# Discovery
+
+Storage nodes must register theselves with the protocol so they can be discovered by new users. These storage nodes will provide global metadata that can assist new users select the most appropriate storage nodes to use.
+
+The protocol will provide a default set of logic to determine the most appropriate storage nodes for a new user, however users will be able to override this and manually select the storage nodes they wish to use. It is, after all, their data.
+
+Self-hosted storage nodes can be created and used by advanced users. They don't need to be registered with Verida. These nodes can be shared privately amongst groups of users, but won't be discoverable by general users of the Verida network.
+
+The discovery process will be implemented in two phases.
+
+## Phase 1
+
+Verida will manually verify storage node operators and maintain a list of trusted storage nodes with accurate metadata about each storage node.
+
+## Phase 2
+
+A Verida smart contract will be deployed that permits any storage node operator to register themselves on the network.
+
+This phase is out of scope for this VIP.
+
+# Selection considerations
+
+The process of selecting appropriate storage nodes requires consideration of:
+
+- Geographic location
+- Availability
+- Performance
+- Regulatory jurisdiction
+- Node operator
+- User preference
+
+The protocol must specify a default selection process for new users. It's intended that advanced users can select their own storage nodes at any time (including selecting self-hosted storage nodes).
+
+## Geographic location
+
+The location of the storage node could consist of the following relevant metadata:
+
+- Country
+- Region (ie: Europe, South East Asia)
+- Datacenter
+- Latitude / Longitude
+
+## Recommendation
+
+1. A registry of all datacenters will be created (`Datacenter Registry`) that has a `uniqueId`, `name`, `countryLocation`, `latitude` and `longitude`.
+2. Each storage node will be included in the `Storage Node Registry` and specify the `uniqueId` of the datacenter where the node is located.
+3. A registry of all regions will be created (`Region Registry`) that maps `country` to `region`. This will be helpful for the client protocol to narrow down the most appropriate storage nodes to consider for selection.
+
+## Availability
+
+The availability of a storage node could consider the following relevant metadata:
+
+- % uptime (ie: 99.99% uptime)
+- Establishment date (ie: created 1 day ago is less trustworthy than created 1,000 days ago)
+- Outages (ie: 5 outages in the last day)
+
+## Recommendation
+
+1. Providing dynamic feedback on uptime and outages is a non-trivial process, so `% uptime` and `outages` is out of scope for now.
+2. `establishmentDate` will be included in the `Storage Node Registry`.
+
+## Performance
+
```

Comment:
It could be good to have a DNS name or IP address here to check. This could be useful for discovery of network proximity like how VPN clients do it

![image](https://user-images.githubusercontent.com/87622993/210452975-0b256042-631b-4237-8020-6ebc32f50b9d.png)

Reply:
Or maybe the intention here is to use the `serviceEndpoint`?

Reply:
Yes, it will use `serviceEndpoint`.